src/services/resource-creator.js

```diff
@@ -5,55 +5,85 @@ const { ErrorHTTP422 } = require('./errors');
 const ResourceGetter = require('./resource-getter');
 const CompositeKeysManager = require('./composite-keys-manager');
 
-function ResourceCreator(model, params) {
-  const schema = Interface.Schemas.schemas[model.name];
-
-  this.perform = function perform() {
-    const promises = [];
-    const recordCreated = model.build(params);
-
-    if (model.associations) {
-      _.forOwn(model.associations, (association, name) => {
-        if (association.associationType === 'BelongsTo') {
-          promises.push(recordCreated[`set${_.upperFirst(name)}`](params[name], { save: false }));
-        }
-      });
+const getSetterName = (associationType, name) => {
+  let setterName;
+  if (associationType === 'HasOne') {
+    setterName = `set${_.upperFirst(name)}`;
+  } else if (['BelongsToMany', 'HasMany'].includes(associationType)) {
+    setterName = `add${_.upperFirst(name)}`;
+  }
```

Comment:
I still feel unconfortable with unreachable code.
Maybe you can merge this inside `getPromisesAfterSave`

Reply:
I will try :)
Hoping codeclimate is ok with that 🤞